metricbeat/module/kubernetes/apiserver/metricset.go

```diff
@@ -29,7 +31,7 @@ type metricset struct {
 	prometheusMappings *prometheus.MetricsMapping
 }
 
-var _ mb.ReportingMetricSetV2 = (*metricset)(nil)
+var _ mb.ReportingMetricSetV2Error = (*metricset)(nil)
```

Comment:
What is this for? to check that `metricset` implements the interface? It may worth a comment.

Reply:
I honestly have no idea, I couldn't figure it out. All I know is that it wouldn't compile without it.

Reply:
Some sort of compile-time check to make sure `metricset` implements the interface is probably the answer. It's a...tad odd.

Reply:
Ok, let's leave it as is, it seems that this pattern is not so uncommon.